### ISSUE TITLE
Video recording option

### ISSUE THREAD

**Issue by marespiaut** [chocolate-doom/chocolate-doom#715]
Hello,

I would like to know if a lossless video recording option would fit in the philosophy of Crispy Doom.
It would be interesting to have a launch parameter  `--record-video` or a key-press in-game to start the recording of the game into a lossless video file. Something like DOSBox have.

**Comment by fabiangreffrath** [comment]
This idea isn't too far-fetched. In fact, there has already been some effort to support video recording from `-timedemo` playbacks in the viddump branch:
https://github.com/chocolate-doom/chocolate-doom/tree/viddump

However, it has some issues such as depending on the whole FFMpeg library stack, not being able to record sound (the recordings are mute) and no explicit selection of video codecs and encoding parameters. Any help on this front is very welcome.

**Comment by Danfun64** [comment]
How about the ability to record video and audio live, especially for during multiplayer matches (demos don't preserve lag frames or text messages, both fairly useful IMO).

**Comment by CraftedPixel** [comment]
There is demo recording.
But that is "beyond the scope of the project" - fragglet
But you could use video recording software.
By the way. Lossless video.... Prepare at least 1000 Terrabytes.

**Comment by marespiaut** [comment]
>  By the way. Lossless video.... Prepare at least 1000 Terrabytes.

@CraftedPixel No, video codecs such as [ZMBV](https://wiki.multimedia.cx/?title=DosBox_Capture_Codec), [H.264](https://trac.ffmpeg.org/wiki/Encode/H.264#LosslessH.264) and [H.265](http://x265.readthedocs.io/en/default/lossless.html) can encode lossless quality video and significantely reduce the file size compared to a raw source.

**Comment by CraftedPixel** [comment]
@maresplaut Are you thinking of publishing it on youtube? Youtube will compress it anyway.
Here is something to think about.

Was there a video recording software with the vanilla doom?
If not I suggest you close this issue.

You should just record using your recording software.

**Comment by fragglet** [comment]
@CraftedPixel - to my knowledge I have never said that video recording is "beyond the scope of the project". Can you cite where you believe I've said that? If not then please don't put words in my mouth.

To state explicitly, video recording for Chocolate Doom is perfectly within scope of the project. We have an implementation on viddump branch but it is not yet merged. I want to ensure we can do both live recordings and batch conversions of demo .lmps.

**Comment by AXDOOMER** [comment]
Splitscreen support would be great too. It would be a simple way to record multiplayer demos (for youtube) and show every player on the screen at the same time. 

It's too complicated to do it with programs like Sony Vegas. I've attempted this before. The sound from each player's screen plays above each other and it sounds like garbage.

**Comment by jmtd** [comment]
> It's too complicated to do it with programs like Sony Vegas. I've attempted this before. The sound from each player's screen plays above each other and it sounds like garbage.

How should the sound work in a split-screen situation? I'm thinking of a sound effect that more than one player experiences, like a monster wake event. Play the sound effect twice, simultaneously, with the right volume, panning etc. for each player, or play one instance, with some kind of blended volume/panning calculation?

**Comment by AXDOOMER** [comment]
Don't play it twice, just play the louder one. We should find out how the Xbox port does it or try some options until we find one that works better than the others.

**Comment by fragglet** [comment]
Splitscreen support sounds like it would require splitscreen rendering to achieve. I'm sorry that the program you used didn't work well but I don't think this is in scope for Chocolate Doom. It's best if we try to just do one thing well.

**Comment by fx02** [comment]
That's good idea for crispy doom.

**Comment by linguica** [comment]
If you want to record a multiplayer vanilla demo in splitscreen, may I suggest https://www.doomworld.com/vb/eternity/91789-splitscreen-branch-downloadable-builds-v0-11/ ?

**Comment by AXDOOMER** [comment]
@linguica  Thanks for the suggestion.

**Comment by fabiangreffrath** [comment]
> That's good idea for crispy doom.

It's just as good for Crispy as it is for Choco, or not?

**Comment by PorscheMonty** [comment]
One of the most frustrating aspects of recording video for vanilla Doom has been getting the app to stick to 35fps. What we get instead is 60, 70 or whatever fps video because the recording is done according to the refresh rate of either the host or the guest machine;  emulators have no real concept of internal fps (naturally). Sometimes it's possible to choose the capture rate, but more often than not it will still require as much postprocessing as anything else to get rid of the irregular motion.

That's one thing I'd love to see :)
